core-management/src/main/java/org/wildfly/extension/core/management/ConfigurationChangeResourceDefinition.java

```diff
@@ -109,32 +97,49 @@ public void registerAttributes(ManagementResourceRegistration resourceRegistrati
         resourceRegistration.registerReadWriteAttribute(MAX_HISTORY, null, new MaxHistoryWriteHandler(ConfigurationChangesCollector.INSTANCE));
     }
 
+
     private static class ConfigurationChangeResourceAddHandler extends AbstractAddStepHandler {
-        public ConfigurationChangeResourceAddHandler() {
-            super(MAX_HISTORY);
+        private ConfigurationChangeResourceAddHandler() {
+            super(CONFIGURATION_CHANGES_CAPABILITY, MAX_HISTORY);
+        }
+
+        @Override
+        protected void performRuntime(OperationContext context, ModelNode operation, Resource resource) throws OperationFailedException {
+            super.performRuntime(context, operation, resource);
+            context.getServiceTarget().addService(CONFIGURATION_CHANGES_CAPABILITY.getCapabilityServiceName(), new AbstractService<Void>() {}).install();
+            ModelNode maxHistory = MAX_HISTORY.resolveModelAttribute(context, operation);
+            ConfigurationChangesCollector.INSTANCE.setMaxHistory(maxHistory.asInt());
+        }
+
+        @Override
+        protected boolean requiresRuntime(OperationContext context) {
+            return context.isDefaultRequiresRuntime() && context.getRunningMode() != RunningMode.ADMIN_ONLY;
         }
 
         @Override
         protected void populateModel(OperationContext context, ModelNode operation, Resource resource) throws OperationFailedException {
             super.populateModel(context, operation, resource);
-            ModelNode maxHistory = MAX_HISTORY.resolveModelAttribute(context, operation);
             MAX_HISTORY.validateAndSet(operation, context.readResourceForUpdate(PathAddress.EMPTY_ADDRESS).getModel());
```

Comment:
This shouldn't be needed as you pass MAX_HISTORY to the superclass constructor. So the whole method can go.